zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/UserTaskVariableHandler.java

```diff
@@ -54,8 +53,18 @@ public boolean handlesRecord(final Record<VariableRecordValue> record) {
 
   @Override
   public List<String> generateIds(final Record<VariableRecordValue> record) {
-    return List.of(
-        ID_PATTERN.formatted(record.getValue().getScopeKey(), record.getValue().getName()));
+    final String id =
+        ID_PATTERN.formatted(record.getValue().getScopeKey(), record.getValue().getName());
+
+    /* Process Variable */
+    if (record.getValue().getScopeKey() == record.getValue().getProcessInstanceKey()) {
+      return List.of(id);
+    }
```

Comment:
🤔 Not sure whether this is correct.

If the scope is a sub-process, this will create a local variable. But we want only local variables for user tasks or? 🤔

Reply:
That is true, I've tested a sub-process scenario and it generates local variables as well, good catch! I'll have to adjust this.

Reply:
The problem is that at this moment, it is not known if the scope is a subprocess or not. So, while I would agree to avoid unnecessary duplications, for now, there is no such information (quickly) available to filter by. So, IMO, that's actually expected and okay.